src/crimson/osd/pg.cc

```diff
@@ -434,7 +434,7 @@ PG::do_delete_work(ceph::os::Transaction &t, ghobject_t _next)
     t.remove(coll_ref->get_cid(), pgmeta_oid);
     t.remove_collection(coll_ref->get_cid());
     (void) shard_services.get_store().do_transaction(
-      coll_ref, std::move(t)).then([this] {
```

Comment:
I'm not following the difference between moving `t` and calling `t.claim_and_reset()` (which looks to be a swap() internally).

Appriciated if you can help elaborate this.

Reply:
See https://github.com/ceph/ceph/pull/55305#issuecomment-1910350264
we move `t` and then use it again without resetting. `claim_and_reset` will move **and** reset the txn.

Reply:
Oh, I think you mean that `ShardServices::dispatch_context_transaction` will reuse the same transaction data-structure after `PG::do_delete_work`. UB will happen if it is not reset. Thanks!